Mailbox/DataLoader.cs

```diff
@@ -1,21 +1,83 @@
-﻿using System.Collections.Generic;
+﻿using Newtonsoft.Json;
+using System;
+using System.Collections.Generic;
 using System.IO;
+using System.Text;
 
 namespace Mailbox
 {
-    public class DataLoader
+    public class DataLoader : IDisposable
     {
+        public Stream Source { get; set; }
+
         public DataLoader(Stream source)
         {
+            Source = source;
         }
 
-        public List<Mailbox> Load()
+        public List<Mailbox>? Load()
         {
+            List<Mailbox> mailboxes = new List<Mailbox>();
+            Source.Seek(0, SeekOrigin.Begin);
+            using StreamReader streamReader = new StreamReader(Source, new UTF8Encoding(), false, 1024, true);
+            while (!(streamReader.EndOfStream))
+            {
+                string mailboxString = streamReader.ReadLine() ?? "";
+                Mailbox mailbox = new Mailbox(Sizes.Default, (0, 0), new Person("", ""));
+                try
+                {
+                    mailbox = JsonConvert.DeserializeObject<Mailbox>(mailboxString);
+                }
+                catch (JsonReaderException jsonReaderException)
+                {
+                    Console.WriteLine(nameof(jsonReaderException));
```

Comment:
Not sure this WriteLine adds much value here. Perhaps
```suggestion
                    Console.WriteLine(jsonReaderException.ToString());
```